### PR TITLE
Re-integrate simple gui to action server

### PR DESCRIPTION
**Pull request by AustinDeric**
@woodaaron made some action servers in #200.  This PR enables the simple gui to work with these actions.

@jrgnicho please review and merge.

### CODE REVIEW

File: godel_surface_detection/src/detection/surface_detection.cpp [hunk 1]
```diff
@@ -433,5 +410,29 @@ namespace godel_surface_detection
...
+      pass.filter (*intermediate_cloud_ptr);
+
+      //downsample the full cloud using the voxelgrid filter method
+      pcl::VoxelGrid<pcl::PointXYZRGB> vox;
+      vox.setInputCloud (intermediate_cloud_ptr);
+      vox.setLeafSize (INPUT_CLOUD_VOXEL_FILTER_SIZE,
+                       INPUT_CLOUD_VOXEL_FILTER_SIZE,
+                       INPUT_CLOUD_VOXEL_FILTER_SIZE);
+      vox.filter(*process_cloud_ptr_);
+    }
```
Comment (jrgnicho) [inline godel_surface_detection/src/detection/surface_detection.cpp:436]: This approach assumes that the table is very close to the robot base z = 0 , is this meant to be a permanent solution or just a temporary workaround? It seems that the findLargestCloud() method would've accomplished the same table removal task without making any assumptions about the tables z placement.

File: godel_simple_gui/src/states/simulating_state.cpp [hunk 2]
```diff
@@ -23,7 +19,7 @@ void godel_simple_gui::SimulatingState::onStart(BlendingWidget& gui)
 
   sim_client_ =
       gui.nodeHandle().serviceClient<godel_msgs::SelectMotionPlan>(SELECT_MOTION_PLAN_SERVICE);
-  QtConcurrent::run(this, &SimulatingState::simulateAll);
+  QtConcurrent::run(this, &SimulatingState::simulateAll, boost::ref(gui));
 }
```
Comment (jrgnicho) [inline godel_simple_gui/src/states/simulating_state.cpp:23]: Why do wee need to pass a pointer to the gui now?

File: godel_surface_detection/src/detection/surface_detection.cpp (hunk 1 above)
Comment (AustinDeric) [reply godel_surface_detection/src/detection/surface_detection.cpp:436]: We chose this method for speed.  It is only valid,as you say, when "robot base z = 0" .

File: godel_simple_gui/src/states/simulating_state.cpp (hunk 2 above)
Comment (AustinDeric) [reply godel_simple_gui/src/states/simulating_state.cpp:23]: the action server is contained in the gui object as a public function, therefore we need to pass the object to use it.  This is the approach of the simple gui.

File: godel_surface_detection/src/detection/surface_detection.cpp (hunk 1 above)
Comment (jrgnicho) [reply godel_surface_detection/src/detection/surface_detection.cpp:436]: If this is meant to be a quick workaround for expedience then I think this should this be logged as an enhancement/issue that we can address in the future?

**Comment by jrgnicho** [comment]
@AustinDeric  just in case, were you able to test the GUI changes? if so then feel free to merge.  Thanks

**Pull request merged**